pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -288,6 +280,18 @@ func (cd *clusterDeploy) deployAgent(cluster *apimgmtv3.Cluster) error {
 	}
 	logrus.Tracef("clusterDeploy: deployAgent: desiredTaints is [%v] for cluster [%s]", desiredTaints, cluster.Name)
 
+	// add additional tolerations and override affinity/resource limits if user has specified. If affinity is not
+	// overridden, use default.
+	tolerations := util.GetTolerations(cluster, desiredTaints)
+	logrus.Infof("clusterDeploy: deployAgent: tolerations are [%v] for cluster [%s]", tolerations, cluster.Name)
```

Comment:
Do we want these logs on `Infof` or `Tracef` like the log above?

Reply:
NA because I'm moving that util logic into SystemTemplate/import so these log messages will be removed

Reply:
@a-blender what did you mean by moving this to systemtemplate? you'll need to get them in `redeployAgent` function to check if spec vs status changed for redeploying agent...I see that you have it as todo, but based on last discussion with @snasovich, we're still allowing agents to redeploy when these fields change on cluster sync. 

It'd make sense to create a struct or common function to get all desired fields, send it to `redeployAgent` to check if redeploy is needed and then use it in `getYAML` func.

Reply:
@kinarashah That was my gut reaction to that comment, that we'd still need to access the fields in `redeployAgent` but we have a reference to the v3 cluster there and access to my new utils function which I will push shortly. The refactor will be up in a minute

Reply:
@kinarashah , this was done in response to my earlier comment: https://github.com/rancher/rancher/pull/41203#discussion_r1173891128

Not taking `redeloyAgent` check I was thinking we want to move calculation of resource limits, affinity, and additional tolerations to `SystemTemplate` func since it already receives `v3.Cluster` object instead of adding additional fields to the already long func signature and duplicating the logic to resolve values for all these fields in multiple places.
Maybe I'm missing some reason why it may be a bad idea like how much logic we want in `SystemTemplate` function - there doesn't seem to be clear definition of how "dumb" or "smart" this function should be by design.

We can discuss in standup afterparty if you want to.

cc: @a-blender

Reply:
@snasovich I agree with what you said and like the idea of making it cleaner. As long as we can access the same fields in redeployAgent, we should be good. Apologies for the confusion.